src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -1171,7 +1179,7 @@ public void refreshUserAppsFullNameValues() {
 
             // get entity key ids
             Set<String> fullNames = StreamUtil.stream( data )
-                    .map( entry -> entry.get( FULL_NAME_FQN ).iterator().next().toString() )
+                    .map( entry -> getFirstValueOrNull( entry, FULL_NAME_FQN ) )
```

Comment:
you probably want to filter out the nulls after this, i.e.
```
.map( ... )
.filter( value -> value != null)
.collect( ... )
```

(although someday when this is moved over to kotlin you can just use kotlin's built-in `.mapNotNull()` function :) )

Reply:
Updated.